services/venice-controller/src/main/java/com/linkedin/venice/controller/VeniceParentHelixAdmin.java

```diff
@@ -3987,10 +4089,20 @@ private void handleTerminalJobStatus(
 
     boolean isTargetRegionPush = !StringUtils.isEmpty(targetedRegions);
 
-    if (!isTargetRegionPush // Push is complete for a normal batch push w/o target region push
-        || isPushCompleteInAllRegionsForTargetRegionPush // Push is complete in all regions for a target region push w/o
-                                                         // deferred swap
-        || isHybridStore // Push is to a hybrid store
+    if (multiClusterConfigs.getConcurrentPushDetectionStrategy().isTopicWriteNeeded() && (!isTargetRegionPush // Push is
```

Comment:
Can you move comments on the separate line? 

Also, IMO we shouldn't touch truncation code yet. Since we could potentially leak topics if version based tracking was enabled after the topic was created but before it was deleted